blocks/gu_spdetails/block_gu_spdetails.php

```diff
@@ -408,7 +416,8 @@ public static function return_feedback($feedback, $gradingduedate) {
 
         $feedbackrecord->hasurl = $feedback ? true : false;
         $feedbackrecord->text = $feedback ? get_string('readfeedback', $lang) :
-                    get_string('due', $lang).userdate($gradingduedate,  get_string('convertdate', $lang));
+                                $gradingduedate == 0 ? '-' : get_string('due', $lang).userdate($gradingduedate,
```

Comment:
Fix on logic needed: Assessments with feedback displays `-` instead of `Read feedback` .

Reply:
Fix on logic resolved on recent push